pkg/controller/build/build_controller.go

```diff
@@ -992,121 +1041,116 @@ func (ctrl *Controller) prepareForBuild(inputs *buildInputs) (ImageBuildRequest,
-// build, and updates the MachineConfigPool with an object reference for the
-// build pod.
-func (ctrl *Controller) startBuildForMachineConfigPool(ps *poolState) error {
-
-	if ctrlcommon.DoARebuild(ps.pool) {
-		// delete FAILED build attempts and builds
-		// delete rendered containerfile, MC, configmaps etc.
-		// Delete the actual build object itself.
-		// if we are rebuilding, we cannot ignore DNE. All of these objects should exist.
-		err := ctrl.postBuildCleanup(ps.pool, false)
-		if err != nil {
-			return fmt.Errorf("Could not update pool when triggering a rebuild: %v", err)
-		}
-		// remove annotation
-		delete(ps.pool.Labels, ctrlcommon.RebuildPoolLabel)
-		err = ctrl.updatePoolAndSyncAvailableStatus(ps.MachineConfigPool())
-		if err != nil {
-			return fmt.Errorf("Could not update pool when triggering a rebuild: %v", err)
-		}
+// Fetches an optional secret to inject into the build. Returns a nil error if
+// the secret is not found.
+func (ctrl *Controller) getOptionalSecret(secretName string) (*corev1.Secret, error) {
+	optionalSecret, err := ctrl.kubeclient.CoreV1().Secrets(ctrlcommon.MCONamespace).Get(context.TODO(), secretName, metav1.GetOptions{})
+	if err == nil {
+		klog.Infof("Optional build secret %q found, will include in build", secretName)
+		return optionalSecret, nil
+	}
 
+	if k8serrors.IsNotFound(err) {
+		klog.Infof("Could not find optional secret %q, will not include in build", secretName)
+		return nil, nil
 	}
-	inputs, err := ctrl.getBuildInputs(ps)
-	if err != nil {
-		return fmt.Errorf("could not fetch build inputs: %w", err)
+
+	return nil, fmt.Errorf("could not retrieve optional secret: %s: %w", secretName, err)
+}
+
+// Fetches an optional ConfigMap to inject into the build. Returns a nil error if
+// the ConfigMap is not found.
+func (ctrl *Controller) getOptionalConfigMap(configmapName string) (*corev1.ConfigMap, error) {
+	optionalConfigMap, err := ctrl.kubeclient.CoreV1().ConfigMaps(ctrlcommon.MCONamespace).Get(context.TODO(), configmapName, metav1.GetOptions{})
+	if err == nil {
+		klog.Infof("Optional build ConfigMap %q found, will include in build", configmapName)
+		return optionalConfigMap, nil
 	}
 
-	ibr, err := ctrl.prepareForBuild(inputs)
-	if err != nil {
-		return fmt.Errorf("could not start build for MachineConfigPool %s: %w", ps.Name(), err)
+	if k8serrors.IsNotFound(err) {
+		klog.Infof("Could not find ConfigMap %q, will not include in build", configmapName)
+		return nil, nil
 	}
 
-	objRef, err := ctrl.imageBuilder.StartBuild(ibr)
+	return nil, fmt.Errorf("could not retrieve optional ConfigMap: %s: %w", configmapName, err)
+}
 
+// Determines if we should run a build, then starts a build pod to perform the
+// build, and updates the MachineConfigPool with an object reference for the
+// build pod.
+func (ctrl *Controller) startBuildForMachineConfigPool(mosc *mcfgv1alpha1.MachineOSConfig, mosb *mcfgv1alpha1.MachineOSBuild) error {
```

Comment:
suggestion (non-blocking): We should change the name of this method to something like `startBuild` or similar since the MachineConfigPool is no longer the source of truth for build state.

(We can address this later though.)